deploy/eck-integrations/templates/eck-agent/_kubernetes/_kubernetes_metrics_apiserver.tpl

```diff
@@ -0,0 +1,41 @@
+{{/*
+Config input for kube apiserver
+*/}}
+{{- define "agent.kubernetes.config.kube_apiserver.input" -}}
+{{- if default .control_plane.apiserver.enabled false -}}
+- id: kubernetes/kubernetes/metrics-kube-apiserver
+  type: kubernetes/metrics
+  data_stream:
+      namespace: {{ .namespace }}
+  use_output: default
+  streams:
+  - id: kubernetes/metrics-kubernetes.apiserver
+    data_stream:
+        type: metrics
+        dataset: kubernetes.apiserver
+    metricsets:
+        - apiserver
+{{- include "agent.kubernetes.config.kube_apiserver.defaults" .control_plane.apiserver | nindent 4 }}
+  meta:
+    package:
+      name: kubernetes
+      version: {{ .version }}
+{{- end -}}
+{{- end -}}
+
+
+{{/*
+Defaults for kube_apiserver input streams
+*/}}
+{{- define "agent.kubernetes.config.kube_apiserver.defaults" -}}
+hosts:
+{{- range dig "vars" "hosts" (list "https://${env.KUBERNETES_SERVICE_HOST}:${env.KUBERNETES_SERVICE_PORT}") . }}
+- {{. | quote}}
+{{- end }}
+period: {{ dig "vars" "period" "30s" . }}
```

Comment:
short preview but I will update just one entry with comments in values.yaml to have this there as well

```yaml
kubernetes:
  objects:
    pods:
      state:
        vars:
          add_metadata: false
          hosts:
          - "localhost: 5050"
```

as an example above this will affect only the config of kube_state for pods and apply the values under `vars` instead of the default ones